tests/lib/schema_validation_test.py

```diff
@@ -151,6 +154,57 @@ def test_valid_without_auto_neg_and_missing_speed(self, default_data):
         libnmstate.validator.validate(default_data)
 
 
+class TestIfaceTypeTeam(object):
+    def test_valid_team_without_options(self, default_data):
+        default_data[Interface.KEY].append(
+            {
+                Interface.NAME: TEAM0,
+                Interface.TYPE: Team.TYPE,
+                Team.CONFIG_SUBTREE: {},
+            }
+        )
+
+        libnmstate.validator.validate(default_data)
+
+    def test_valid_team_without_config(self, default_data):
+        default_data[Interface.KEY].append(
+            {
+                Interface.NAME: TEAM0,
+                Interface.TYPE: Team.TYPE,
+                Team.CONFIG_SUBTREE: {Team.SLAVES: ['eth1', 'eth2']},
+            }
+        )
+
+        libnmstate.validator.validate(default_data)
+
+    def test_valid_team_without_slaves(self, default_data):
+        default_data[Interface.KEY].append(
+            {
+                Interface.NAME: TEAM0,
+                Interface.TYPE: Team.TYPE,
+                Team.CONFIG_SUBTREE: {
+                    Team.CONFIG: '{\"runner\": {\"name\": \"broadcast\"}}'
```

Comment:
I know that team API uses plain JSON to carry configuration, however, that should not be exposed in nmstate.
The API here should be well defined and part of the YAML schema.

Would it be possible to expose these JSON options (or a subset of them) explicitly in the schema?

Reply:
Yes, it is possible but then in the implementation we would need to convert it to plain JSON. Is that fine? If it is, I will work on that.

Reply:
I think that's all right. We should make it simple and pretty for nmstate users, hide JSON config from them. Selecting specific configuration options could also help us identify "how" will be team interfaces typically used.

Reply:
Yes, this approach is more user friendly. Thanks!